modules/selection/src/main/kotlin/com.r3.corda.lib.tokens.selection/memory/services/VaultWatcherService.kt

```diff
@@ -279,8 +300,14 @@ class VaultWatcherService(private val tokenObserver: TokenObserver,
         indexViewCreationLock.read {
             for (stateAndRef in stateAndRefs) {
                 val existingMark = __backingMap.remove(stateAndRef)
-                existingMark
-                        ?: LOG.warn("Attempted to remove existing token ${stateAndRef.ref}, but it was not found this suggests incorrect vault behaviours")
+                if (existingMark == null) {
+                    if (tokenLoadingFinished) {
+                        LOG.warn("Attempted to remove existing token ${stateAndRef.ref}, but it was not found this suggests incorrect vault behaviours")
+                    }
+                    else if (stateAndRefs !== missedConsumes){
```

Comment:
`tokenLoadingStarted && stateAndRefs !== missedConsumes` 

Again for correctness when you might be using the other constructor in tests.